libminifi/test/unit/TimeUtilTests.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utils/TimeUtil.h"
+#include "../TestBase.h"
+
+namespace {
+  constexpr int ONE_HOUR = 60 * 60;
+  constexpr int ONE_DAY = 24 * ONE_HOUR;
+
+  struct tm createTm(int year, int month, int day, int hour, int minute, int second, bool is_dst = false) {
+    struct tm date_time;
+    date_time.tm_year = year - 1900;
+    date_time.tm_mon = month - 1;
```

Comment:
Do I understand correctly that months are stored in [0-11] range, but days are stored in [1-31]?

Reply:
Yes, and years start at 1900 -- the person who designed `struct tm` was probably drunk at the time.

Reply:
🍺